File: dvbViewerController/src/main/java/org/dvbviewer/controller/data/media/MediaViewModelFactory.java
package org.dvbviewer.controller.data.media;

import android.app.Application;
import android.arch.lifecycle.ViewModel;
import android.arch.lifecycle.ViewModelProvider;

public class MediaViewModelFactory extends ViewModelProvider.NewInstanceFactory {

    private Application mApplication;
    private MediaRepository mRepo;

    public MediaViewModelFactory(Application application, MediaRepository repo) {
        mApplication = application;
        mRepo = repo;
    }

    @Override
    public <T extends ViewModel> T create(Class<T> modelClass) {
        return (T) new MediaViewModel(mApplication, mRepo);
    }

}

File: dvbViewerController/src/main/java/org/dvbviewer/controller/data/media/MediaViewModelFactory.kt
package org.dvbviewer.controller.data.media

import android.app.Application
import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider

class MediaViewModelFactory(private val mApplication: Application, private val mRepo: MediaRepository) : ViewModelProvider.NewInstanceFactory() {

    override fun <T : ViewModel> create(modelClass: Class<T>): T {
        return MediaViewModel(mApplication, mRepo) as T
    }

}